src/core/include/openvino/core/attribute_adapter.hpp

```diff
@@ -431,22 +437,24 @@ class OPENVINO_API AttributeAdapter<std::vector<uint64_t>> : public DirectValueA
     OPENVINO_RTTI("AttributeAdapter<std::vector<uint64_t>");
 };
 
-/// \brief Access a vector<float>
+/// \brief Access a vector<double>
 template <>
-class OPENVINO_API AttributeAdapter<std::vector<float>> : public DirectValueAccessor<std::vector<float>> {
+class OPENVINO_API AttributeAdapter<std::vector<double>> : public DirectValueAccessor<std::vector<double>> {
 public:
-    AttributeAdapter(std::vector<float>& value) : DirectValueAccessor<std::vector<float>>(value) {}
+    AttributeAdapter(std::vector<double>& value) : DirectValueAccessor<std::vector<double>>(value) {}
 
-    OPENVINO_RTTI("AttributeAdapter<std::vector<float>");
+    OPENVINO_RTTI("AttributeAdapter<std::vector<double>");
 };
 
-/// \brief Access a vector<double>
+/// \brief Access a vector<float>
 template <>
-class OPENVINO_API AttributeAdapter<std::vector<double>> : public DirectValueAccessor<std::vector<double>> {
+class OPENVINO_API AttributeAdapter<std::vector<float>>
+    : public IndirectVectorValueAccessor<std::vector<float>, std::vector<double>> {
 public:
-    AttributeAdapter(std::vector<double>& value) : DirectValueAccessor<std::vector<double>>(value) {}
+    AttributeAdapter(std::vector<float>& value)
+        : IndirectVectorValueAccessor<std::vector<float>, std::vector<double>>(value) {}
 
-    OPENVINO_RTTI("AttributeAdapter<std::vector<double>");
+    OPENVINO_RTTI("AttributeAdapter<std::vector<float>");
 };
 
```

Comment:
looks like nothing has changed in this file? Just reshuffle?

Reply:
DirectValueAccessor -> IndirectValueAccessor
due to protobuf stores attributes values as int64, double,  it will allow us to cast them to int32, float